llvm/unittests/IR/ConstantRangeListTest.cpp

```diff
@@ -0,0 +1,62 @@
+//===- ConstantRangeListTest.cpp - ConstantRangeList tests ----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/IR/ConstantRangeList.h"
+#include "gtest/gtest.h"
+
+using namespace llvm;
+
+namespace {
+
+using ConstantRangeListTest = ::testing::Test;
+
+TEST_F(ConstantRangeListTest, Basics) {
```

Comment:
can you add a test for (0,4) + (4,8)?

Reply:
Nice catch! Changed one "slt" to "sle" in CRL::insert() and updated the unit test to cover this consecutive case.